### PR TITLE
feat: posthog app added

### LINKED ISSUE #13546: App/Analytics: Posthog Support
### Is your proposal related to a problem?

A lot of startups and many other companies use [**Posthog**](https://posthog.com/) for analytics and product measurements. It's also open source!

I'd be great to be able to track conversion on booking pages with it.

### Describe the solution you'd like

The solution would be something along the lines of:

```js
import posthog from 'posthog-js'
posthog.init([TRACKING-ID], { api_host: [OPTIONAL_HOST]' })
```
Alternatively:
```html
<script>
    !function(t,e){var o,n,p,r;e.__SV||(window.posthog=e,e._i=[],e.init=function(i,s,a){function g(t,e){var o=e.split(".");2==o.length&&(t=t[o[0]],e=o[1]),t[e]=function(){t.push([e].concat(Array.prototype.slice.call(arguments,0)))}}(p=t.createElement("script")).type="text/javascript",p.async=!0,p.src=s.api_host+"/static/array.js",(r=t.getElementsByTagName("script")[0]).parentNode.insertBefore(p,r);var u=e;for(void 0!==a?u=e[a]=[]:a="posthog",u.people=u.people||[],u.toString=function(t){var e="posthog";return"posthog"!==a&&(e+="."+a),t||(e+=" (stub)"),e},u.people.toString=function(){return u.toString(1)+".people (stub)"},o="capture identify alias people.set people.set_once set_config register register_once unregister opt_out_capturing has_opted_out_capturing opt_in_capturing reset isFeatureEnabled onFeatureFlags getFeatureFlag getFeatureFlagPayload reloadFeatureFlags group updateEarlyAccessFeatureEnrollment getEarlyAccessFeatures getActiveMatchingSurveys getSurveys".split(" "),n=0;n<o.length;n++)g(u,o[n]);e._i.push([i,s,a])},e.__SV=1)}(document,window.posthog||[]);
    posthog.init([TRACKING-ID], {api_host: [OPTIONAL_HOST]})
</script>
```
*Note: as Posthog is OSS and also supports custom domains in the hosted version, it's important to allow setting an (optional) host.*

If you want to be extra fancy you could also trigger an event when a call is booked:
```js
posthog.capture('calcom_event_booked', {[..optional metadata});
```

### Describe alternatives you've considered

N/A

### Additional context

N/A

### Requirement/Document

[Posthog Docs](https://posthog.com/docs/libraries/js)

### PR DESCRIPTION
**Pull request by vikaspatil0021**
This PR is to add Posthog analytics in the app store

fixes: #13546 

loom video:
https://www.loom.com/share/a7e6dc1389c54886b9bc627440d463bb

## Type of change

- New feature (non-breaking change which adds functionality)

## How should this be tested?

- Add Posthog analytics application to an event type.
- Open up the event page and it should automatically track it.

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/app-store/posthog/package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  "name": "@calcom/posthog",
+  "version": "0.0.0",
+  "main": "./index.ts",
+  "dependencies": {
+    "@calcom/lib": "*"
+  },
+  "devDependencies": {
+    "@calcom/types": "*"
+  },
+  "description": "PostHog is the all-in-one platform for building better products - with product analytics, feature flags, session recordings, a/b testing, heatmaps, and more."
```
Comment (joeauyeung) [inline packages/app-store/posthog/package.json:13]: Would be nice if there was a link for users to get the `TRACKING_ID` and `API_HOST` variables

File: packages/app-store/posthog/components/EventTypeAppCardInterface.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  return (
+    <AppCard
+      app={app}
+      switchOnClick={(e) => {
+        updateEnabled(e);
+      }}
+      switchChecked={enabled}
+      teamId={eventType.team?.id || undefined}>
+      <div className="flex flex-col gap-2">
+        <TextField
```
Comment (joeauyeung) [inline packages/app-store/posthog/components/EventTypeAppCardInterface.tsx:24]: We should make these required

**Review by joeauyeung — CHANGES_REQUESTED** [review]
Thanks for your contribution @vikaspatil0021! I tested this and it works. Just left two comments. Once these are addressed I can approve.

**Comment by vikaspatil0021** [comment]
@joeauyeung Thanks for the review. Please check the changes

File: packages/app-store/posthog/package.json (hunk 1 above)
Comment (vikaspatil0021) [reply packages/app-store/posthog/package.json:13]: Added the linkin config.json url

**Review by joeauyeung — APPROVED** [review]
LGTM thanks again @vikaspatil0021

**Comment by AustinHay** [comment]
Can somebody from Cal.com describe how the Posthog integration works? There are no documents online - we are trying to record a Posthog capture event from the booking page after a booking completes

**Comment by TyceHerrman** [comment]
Also trying to figure this out but no docs 😭. @vikaspatil0021  or @joeauyeung can you help out?

**Comment by TyceHerrman** [comment]
For anyone else that gets here - Posthog staff responded on their community support forum: https://posthog.com/questions/connecting-cal-com-with-post-hog @AustinHay

**Pull request merged**